Update trial info (#2908)
astronomer/docs: astro/trial.md

## After your trial

After your 14-day trial ends, you can no longer access your Deployments and Workspaces from the Cloud UI. You can still access your user account page and Astronomer support forms.

After your trial ends, all Deployments are automatically deleted. If you created a dedicated cluster, it is also deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:

- Contact [sales](https://astronomer.io/contact/) to request a trial extension.
- Run `astro deployment inspect` with the Astro CLI to save your existing Deployment configuration as a JSON or YAML file. See [Astro CLI command reference](cli/astro-deployment-inspect.md).

> - Open the Cloud UI and schedule office hours with an Astronomer engineer.

Author: ```suggestion
- Open the Cloud UI and [schedule office hours](office-hours.md) with an Astronomer engineer.
```

## After your trial

After your 14-day trial ends, you can no longer access your Deployments and Workspaces from the Cloud UI. You can still access your user account page and Astronomer support forms.

> After your trial ends, your cluster and all Deployments within it are automatically deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:

Reviewer: ```suggestion
After your trial ends, all Deployments are automatically deleted. If you created a dedicated cluster, it is also deleted. Any code that you deployed to Astro will be lost. If you need additional time to evaluate Astro, or you need to copy your configuration for future use, you can:
```

- Schedule a call with [sales](https://astronomer.io/contact/).

> - Open the Cloud UI and schedule a 15 minute call with an Astronomer engineer. This option is available only after your 14-day trial ends.

Reviewer: Is this true?

Author: I can also get Nick to do a review pass, but that's how I understood this Slack explanation - https://astronomer.slack.com/archives/C01UA03TPV1/p1696301864953699?thread_ts=1696286609.279489&cid=C01UA03TPV1

> - Schedule a call with [sales](https://astronomer.io/contact/).
> - Open the Cloud UI and schedule a 15 minute call with an Astronomer engineer. This option is available only after your 14-day trial ends.

Reviewer: ```suggestion
- Contact [sales](https://astronomer.io/contact/) to request a trial extension.
- Run `astro deployment inspect` with the Astro CLI to save your existing Deployment configuration as a JSON or YAML file. See [Astro CLI command reference](cli/astro-deployment-inspect.md).
- Open the Cloud UI and schedule office hours with an Astronomer engineer.
```
